### DISCUSSION TITLE
Scripts which execute only once per collection run

### DISCUSSION THREAD

**Discussion by Tobias4872** [Kong/insomnia#8120] [Help] [answered]
I have started using Insomnia and I really love it. I have been using the pre-request and after-request scripts quite a while now, but what I a missing (or was not able to find yet), is a way to execute a script only once per collection run.

My use case here would be that I need a random number (or uuid or similar) or a date which is generated once and then stays the same across the collection run.

I thought about putting a pre-request script at the top level and check with "insomnia.environment.has()" if a particular variable is set and if not set it. But I also need to clean it afterwards and I failed to find a nice way to do it.

Any help would be greatly appreciated.

**Comment by ihexxa** [comment, marked as answer]
This is an interesting topic 👍 . Hm currently there is no capability for setup or teardown.

One way I can think of is adding 2 requests for them, and calling `insomnia.execution.skipRequest()` to skip sending request from pre-req script, here is an example (setup is similar):
![image](https://github.com/user-attachments/assets/60ed616d-4d93-449f-b76b-5b9ee6ada136)

And set them as the first and the last one.
![image](https://github.com/user-attachments/assets/7fe68e17-2ab6-4836-9b2f-559d19d00aae)

It is still not optimal although it doesn't send out the request.

**Comment by Tobias4872** [comment]
Thanks for the answer. I solved it in a similar fashion. I am just using the first request in each block and add a pre-request script.

The solution with the "skipRequest" is a nice one, though.

**Comment by ihexxa** [comment]
@Tobias4872 10.2 adds support for `insomnia.variables.set` which sets transient variables only valid during collection run. It may be helpful to this case.
